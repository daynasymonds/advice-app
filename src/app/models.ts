export type Slip = {
    slip_id: number;
    advice: string;
}

export type ImageData = {
    name: string;
    alt: string;
}

export type Advice = {
    text: string;
    imageName: string;
    alt: string;
}