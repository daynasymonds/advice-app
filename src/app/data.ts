import { Advice, Slip, ImageData } from "./models";

const images: ImageData[] = [
  { name: "calm-wave.svg", alt: "Calm wave" } as ImageData,
  { name: "rainbow-sun.svg", alt: "Rainbow and sun" } as ImageData,
  { name: "rainbow.svg", alt: "Rainbow" } as ImageData,
  { name: "stars.svg", alt: "Stars" } as ImageData,
  { name: "sunrise.svg", alt: "Sunrise" } as ImageData,
  { name: "sunset.svg", alt: "Sunset" } as ImageData,
];

function randomImangeNameIndex(): number {
  return Math.floor(Math.random() * images.length);
}

export async function getInitialAdvice(): Promise<Advice> {
    return {text: "Always seek out advice or opinions when making a decision.", imageName: "rainbow.svg", alt: "Rainbow"} as Advice;
}

export async function getAdviceData(): Promise<Advice> {
  const adviceData = await fetch("https://api.adviceslip.com/advice");
  const slipObj = await adviceData.json();
  const slip: Slip = slipObj.slip;

  const imageData = images.at(randomImangeNameIndex());

  if (!imageData) {
    throw new Error("Failed to retrieve image data");
  }

  return {
    text: slip.advice,
    imageName: imageData.name,
    alt: imageData.alt,
  } as Advice;
}
