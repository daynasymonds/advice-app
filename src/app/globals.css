@import "tailwindcss";

:root {
  --background: #ffffff;
  --foreground: #171717;
}

@theme inline {
  --color-background: var(--background);
  --color-foreground: var(--foreground);
  --font-montserrat: var(--font-montserrat);
  --color-primary: #295556;
}

body {
  background: var(--background);
  color: var(--foreground);
  font-family: var(--font-montserrat)
}
