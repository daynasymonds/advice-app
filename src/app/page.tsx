
import { Advice } from "@/app/models";
import { getInitialAdvice } from "@/app//data";
import AdviceButton from "@/app/ui/advice-button";

export default async function Home() {
  const initial: Advice = await getInitialAdvice();

  return (
    <div>
      <main>
        <div className="flex h-screen">
            <div className="flex flex-col m-auto px-4 gap-3 md:gap-12 md:px-6">
              <AdviceButton initial={initial}/>
            </div>
        </div>
      </main>
    </div>
  );
}
