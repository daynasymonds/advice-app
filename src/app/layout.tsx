import type { Metadata } from "next";
import { Montserrat } from "next/font/google";
import "./globals.css";


const montserrat = Montserrat({
  variable: "--font-montserrat",
  subsets: ["latin"],
});

export const metadata: Metadata = {
  title: "Advice For Life",
  description: "Dayna Symonds Portfolio App - Simple app demonstrating NextJS/React/TailwindCSS with client and server components.",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <body
        className={`${montserrat.variable}`}
      >
        {children}
      </body>
    </html>
  );
}
