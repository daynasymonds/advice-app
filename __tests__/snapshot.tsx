import { render } from "@testing-library/react";
import AdviceButton from "@/app/ui/advice-button";
import { Advice } from "@/app/models";

it("renders advice button unchanged", () => {
  const initial = {
    text: "a piece of advice",
    imageName: "rainbow.svg",
    alt: "A rainbow",
  } as Advice;

  const { container } = render(<AdviceButton initial={initial} />);
  
  expect(container).toMatchSnapshot();
});
