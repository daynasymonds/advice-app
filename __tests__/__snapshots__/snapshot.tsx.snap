// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders advice button unchanged 1`] = `
<div>
  <p
    class="text-center font-light text-[24px] md:max-w-md] md:text-[36px]"
  >
    a piece of advice
  </p>
  <img
    alt="A rainbow"
    class="h-[128px] w-[128px] self-center md:h-[200px] md:w-[200px]"
    data-nimg="1"
    decoding="async"
    height="128"
    loading="lazy"
    src="rainbow.svg"
    style="color: transparent;"
    width="128"
  />
  <button
    class="rounded-full bg-primary text-white py-1.5 w-[200px] place-self-center font-light duration-[.25s] transition ease-in hover:bg-white hover:border-primary hover:border-solid hover:border-[1px] hover:text-primary"
  >
    Get more advice
  </button>
</div>
`;
